src/scripts/domParsers/khanAcademyVideoExtractor.ts

```diff
@@ -0,0 +1,50 @@
+import {VideoExtractor} from "./videoExtractor";
+import {VideoUtils} from "./videoUtils";
+import {YoutubeVideoExtractor} from "./youtubeVideoExtractor";
+
+import {Utils} from "../utils";
+
+export class KhanAcademyVideoExtractor implements VideoExtractor {
+	private youtubeExtractor: YoutubeVideoExtractor;
```

Comment:
I left it as composition instead of inheritance because I don't think this follows the "is-a" rule of thumb. That being said, it doesn't really follow the "has-a" rule either :P. For inheritance, I also don't make use of a lot of the YouTube code and another good rule of thumb I've read about it is to avoid inheritance unless the vast majority of the code in the superclass is re-used.